launch/process_test.go

```diff
@@ -0,0 +1,227 @@
+							h.AssertNil(t, err)
+							h.AssertEq(t, proc, launch.Process{
+								Command: "user-command",
+								Args:    []string{"user-arg1", "user-arg2"},
+								Direct:  true,
+							})
+						})
+					})
+
+					when("cmd does not starts with --", func() {
+						it("creates a new shell process from cmd", func() {
+							proc, err := launcher.ProcessFor([]string{"user-command", "user-arg1", "user-arg2"})
+							h.AssertNil(t, err)
+							h.AssertEq(t, proc, launch.Process{
+								Command: "user-command",
+								Args:    []string{"user-arg1", "user-arg2"},
+							})
+						})
+					})
+
+					when("cmd is empty", func() {
+						it("errors", func() {
+							_, err := launcher.ProcessFor([]string{})
+							h.AssertNotNil(t, err)
+						})
+					})
+				})
+
+				when("exists", func() {
+					it.Before(func() {
+						launcher.DefaultProcessType = "some-type"
+					})
+
+					it("appends cmd to the default process args", func() {
+						proc, err := launcher.ProcessFor([]string{"user-arg1", "user-arg1"})
+						h.AssertNil(t, err)
+						h.AssertEq(t, proc, launch.Process{
+							Type:        "some-type",
+							Command:     "some-command",
+							Args:        []string{"some-arg1", "some-arg2", "user-arg1", "user-arg1"},
+							BuildpackID: "some-buildpack",
+						})
+					})
+				})
+
+				when("doesn't exit", func() {
+					it.Before(func() {
+						launcher.DefaultProcessType = "missing-type"
+					})
+
+					it("errors", func() {
+						_, err := launcher.ProcessFor([]string{"user-arg1", "user-arg1"})
+						h.AssertNotNil(t, err)
+					})
+				})
+			})
+		})
+
+		when("Platform API < 0.4", func() {
+			it.Before(func() {
+				launcher.PlatformAPI = api.MustParse("0.3")
+			})
+
+			when("DefaultProcessType", func() {
+				when("is unset", func() {
+					when("cmd is empty", func() {
+						it("errors", func() {
+							_, err := launcher.ProcessFor([]string{})
+							h.AssertNotNil(t, err)
+						})
+					})
+
+					when("cmd contains a only a process type", func() {
+						it("returns the process type", func() {
+							proc, err := launcher.ProcessFor([]string{"other-type"})
+							h.AssertNil(t, err)
+							h.AssertEq(t, proc, launch.Process{
+								Type:        "other-type",
+								Command:     "other-command",
+								Args:        []string{"other-arg1", "other-arg2"},
+								BuildpackID: "some-buildpack",
+							})
+						})
+					})
+
+					when("cmd starts with --", func() {
+						it("creates a new direct process from cmd", func() {
+							proc, err := launcher.ProcessFor([]string{"--", "user-command", "user-arg1", "user-arg2"})
+							h.AssertNil(t, err)
+							h.AssertEq(t, proc, launch.Process{
+								Command: "user-command",
+								Args:    []string{"user-arg1", "user-arg2"},
+								Direct:  true,
+							})
+						})
+					})
+
+					when("cmd contains more than one arg and does not start with --", func() {
```

Comment:
I'm a little confused... what happens when the `cmd` contains exactly one arg? Is that covered by these tests?

Reply:
one vs many doesn't matter to `cmd` right now. I'll update the `when`s